### PR TITLE
Fix no undef eslint errors

### PR DESCRIPTION
**Pull request by BCerki**
This PR:

- sets the no undef rules to "error" instead of "off"
- fixes the no undef eslint lint errors
- partially addresses Fixing eslint rule errors Fixing eslint rule errors #37

A couple notes:
- I extended eslint with mocha because we've talked about using some of Quebec's pre-existing mocha tests. If we ultimately don't use it, I can drop it from the eslintrc later.
- There is a [grunt eslint lint extension package](https://www.npmjs.com/package/eslint-config-grunt), but it required some dependencies (?) that overrode the rules in the eslintrc. That seemed like an unnecessary complication, so I chose to add the `gruntConfig` variable as a global instead.

### CODE REVIEW

File: tests/back-end/unit/lib/hooks/logger.ts [hunk 1]
```diff
@@ -35,13 +35,13 @@ function makeStubRequest(onLog: (kind: LogFunctionKind, msg: string, data?: obje
   };
 }
 
-describe('loggerHook', () => {
+describe('loggerHook', function() {
```
Comment (wenzowski) [inline tests/back-end/unit/lib/hooks/logger.ts:38]: Great job finding tooling to enforce [no arrow functions per Mocha guidance](https://mochajs.org/#arrow-functions)!

File: .eslintrc.js [hunk 2]
```diff
@@ -2,12 +2,17 @@ module.exports = {
   "env": {
     "browser": true,
     "es2021": true,
-    "node": true
+    "node": true,
+
+  },
+  "globals": {
+    "gruntConfig": "readonly"
```
Comment (wenzowski) [inline .eslintrc.js:9]: What behaviour does this affect with Grunt?

Comment (BCerki) [reply .eslintrc.js:9]: It fixes this warning/error:
Screenshot from 2021-12-06 15-43-43

Comment (BCerki) [reply .eslintrc.js:9]: Docs on the no-undef error: https://eslint.org/docs/rules/no-undef

Comment (BCerki) [reply .eslintrc.js:9]: Readonly is an eslint thing, not a grunt thing, so it shouldn't have any effect on grunt. When a variable is set to readonly in the globals section, eslint will [give you a warning if you try to change it](https://smartdevpreneur.com/how-and-when-to-specify-globals-in-eslintrc-a-detailed-example-with-configs-and-results/#:~:text=about%20importing%20React.-,Readonly,-and%20Writable%20Globals) (Error: Read-only global 'var1' should not be modified. eslint (no-global-assign)

**Review by mikevespi — APPROVED** [review]
Looks good! 🚢

**Pull request merged**